File: finish_all.c
#include<stdio.h>
#define size 100000
int main()
{
	int n,i,arr[size],j,temp=0,count=0;
	scanf("%d",&n);
	for(i=0;i<n;i++)
	{
		scanf("%d",&arr[i]);	
	}
	for(i=0;i<n;i++)
	{
	if(arr[i]!=-1)
	{
		temp=1;
		for(j=i+1;j<n;j++)
		{
			if(arr[i]==arr[j])
			{
				arr[j]=-1;
				temp++;
			}
		}
		if(temp%2==0)
		{
			count+=(temp/2);
		} 
		else
		{
			count+=((temp+1)/2);
		}
	}	
	}
	printf("%d\n",count);
	return 0;
}

File: finish_all.py
n=int(input())
m={}
arr=list(map(int,input().split()))
count=0
for i in range(len(arr)):
	if arr[i] in m.keys():
		m[arr[i]]+=1
		if m[arr[i]]%2==1:
			count+=1
	else:
		m[arr[i]]=1
		count+=1
print(count)
